kubernetes-docs/sc.md

```diff
@@ -0,0 +1,167 @@
+---
+title: Dynamic Provisioning and StorageClass API
+---
+
+With PV and PVCs one can only provision storage statically i.e. PVs first needs to be created before a Pod claims it. However with StorageClass API Kubernetes enables dynamic volume provisioning which is very unique to Kubernetes. This avoids pre-provisioning of storage and storage is provisioned automatically when a user requests it. 
+ 
+StorageClass API object specifies a provisioner and parameters  which are used to decide which volume plugin to be used and provisioner specific parameters.
+Provisioner could be AWS EBS, vSphere, OpenStack and so on.
+ 
+vSphere is one of the provisioners and it allows following parameters:
+
+**Diskformat** which can be thin(default), zeroedthick and eagerzeroedthick
+
+**Datastores** is an optional field which can be VMFSDatastore or VSANDatastore. This allows user to select the datastore to provision PV from, if not specified the default datastore from vSphere config file is used.
+
+**storagepolicyName** Name of the SPBM policy to be applied. The newly created persistent volume will have the SPBM policy configured with it.
+VSAN storage capability parameters which you can specify explicitly. The newly created persistent volume will have these VSAN storage capabilities configured with it. (TODO - Add link to the section)
+ 
+There are additional parameters which are covered in Storage Policy Management section.
+
+**Note:**
+
+All the example yamls can be found [here](https://github.com/kubernetes/kubernetes/tree/master/examples/volumes/vsphere) unless otherwise specified. Please download these examples.
+
+Let us look at an example of how to use StorageClass for dynamic provisioning
+**Note:** Here you don't need to create vmdk it is created dynamically.
+
+**Create Storage Class**
+
+```
+vpshere-volume-sc-fast.yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: fast
+provisioner: kubernetes.io/vsphere-volume
+parameters:
+    diskformat: zeroedthick
+    fstype:     ext3
+```
+
+You can also specify the datastore in the Storageclass as shown in example 2. The volume will be created on the datastore specified in the storage class. This field is optional. If not specified as shown in example 1, the volume will be created on the datastore specified in the vsphere config file used to initialize the vSphere Cloud Provider.
+
+```
+vsphere-volume-sc-slow.yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: fast
+provisioner: kubernetes.io/vsphere-volume
+parameters:
+    diskformat: zeroedthick
+    datastore: VSANDatastore
+If datastore is member of DataStore Cluster or within some sub folder, the datastore folder path needs to be provided in the datastore parameter as below.
```

Comment:
"If datastore is member of DataStore Cluster or within some sub folder, the datastore folder path needs to be provided in the datastore parameter as below." can be removed. Introduce this as a paragraph before putting the YAML.